### PR TITLE
added in the ability to clone your own listing from the listing card

### LINKED ISSUE #859: Add ability to easily clone a listing
We had it in 1.0 and it's highly requested in 2.0 already. I'll add designs for it when I get a chance and we can work it into our queue.

### PR DESCRIPTION
**Pull request by rmisio**
@morebrownies I jumped the gun on this one. It was requested enough times and the scope was relatively low, so I just knocked it out. Let me know if you think any tweaks are needed.

I just took a guess on the icon:

image

closes #859

### CODE REVIEW

**Comment by morebrownies** [comment]
This feature is awesome. Icon looks good. 👍

File: js/views/ListingCard.js [hunk 1]
```diff
@@ -108,55 +107,63 @@ export default class extends baseVw {
...
   }
 
-  onClickConfirmedDelete() {
-    if (this.destroyRequest && this.destroyRequest.state === 'pending') return false;
+  onClickClone(e) {
+    app.loadingModal.open();
+
+    this.fetchFullListing()
+      .done(xhr => {
+        if (xhr.statusText === 'abort' || this.isRemoved()) return;
```
Comment (jjeffryes) [inline js/views/ListingCard.js]: Same comment here, if the .always callback was added first, couldn't it handle all the checking for abort and isRemoved?

File: js/views/ListingCard.js [hunk 2]
```diff
@@ -108,55 +107,63 @@ export default class extends baseVw {
   onClickEdit(e) {
     app.loadingModal.open();
 
-    const fullListingFetch = this.fullListing.fetch()
-      .done(() => {
-        if (fullListingFetch.statusText === 'abort' || this.isRemoved()) return;
+    this.fetchFullListing()
+      .done(xhr => {
+        if (xhr.statusText === 'abort' || this.isRemoved()) return;
```
Comment (jjeffryes) [inline js/views/ListingCard.js]: If you add the .always callback first, couldn't it handle the checking for abort and isRemoved?

Comment (rmisio) [reply js/views/ListingCard.js]: Even if you opt out early out of the always() callback, that will have no bearing on the done() or fail() callbacks, which will still fire.

Am I misunderstanding you?

Comment (jjeffryes) [reply js/views/ListingCard.js]: No, you're right, I was misthinking thing.

**Pull request merged**